observability/readme.md

```diff
@@ -9,9 +9,64 @@ Building observable systems enables development teams at CSE to measure how well
 3. Measure operational performance of the system.
 4. Identify and diagnose failures to get to the problem fast.
 
+## Pillars of observability
+
+### Logging
+
+Discrete events with the goal of helping engineers identify problem area(s) during failures.
+
+- Pay attention to logging levels. Logging too much will increase costs and decrease application throughtput.
+- Ensure logging configuration can be modified without code changes. Ideally, make it changeable without application restarts.
+- If available, take advantage of logging levels per category allowing granular logging configuration.
+- Check for log levels before logging, thus avoiding allocations and string manipulation costs.
+- Ensure service versions are included in logs, thus having the capability of identifying problematic releases.
+- Log a raised exception only once. In your handlers, only catch expected exceptions that you can handle gracefully, including those for which you would set a specific return code. If you’re only going to log and rethrow (even without clobbering the stack trace), let the top level exception handler get it. If you have cleanup work to do on a catch, do only what is needed and then throw; to maintain the original stack trace. Don’t log a warning or stack trace for expected exceptions. Proper, expected 404s and 403s are not warnings.
+- Fine tune logging levels in production (>= warning for instance). During a new release the verbosity can be increased to facilitate bug identification.
+- If sampling is used, prefer implementing at the service level as opposed to defining in the logging system. This way we can ensure that exceptions (and other important information) will be logged. Another benefit is the reduced amount of roundtrips.
```

Comment:
Little edits in tone: "If using sampling, implement this at the service level rather than defining it in the logging system. This ensures that exceptions (and other important information) is logged. An additional benefit is reduced number of roundtrips."